_wiki/basic-tutorials/logger-configuration.md

```diff
@@ -2,39 +2,64 @@
-## Using SLF4J
+Changing the log level of the fallback logger only has any effect on newly created loggers. Pre-existing loggers will not have their log level changed. So if you want to configure the fallback logger, you should do this as one of the first things in your bot code. If you want to change log levels during runtime, you should use a proper logging framework like Log4j 2 Core that supports this.
+
+All messages of the fallback logger are printed to the standard output stream (`System.out`) and thus usually to your console. If you want to log to a file, database or anything else, you should consider using a proper logging framework which allows you to configure this behavior.
+
+This is how a log line of the fallback logger will look like:
+
+```log
+<time with date            ><level><logger name, usually the logging class              > <message            > <the thread context, here the shard number>
+2018-08-03 20:00:06.080+0200 DEBUG org.javacord.core.util.gateway.DiscordWebSocketAdapter Received HELLO packet {shard=0}
+```
+
+## Using a Proper Logging Framework
 
 ### Adding a Logging Framework
 
-Adding a logging framework of your choice is very straightforward. You can just add it as a dependency and it will be detected by slf4j automatically. This following example adds logback-classic using Maven:
-```xml
-<dependency>
-  <groupId>ch.qos.logback</groupId>
-  <artifactId>logback-classic</artifactId>
-  <version>1.2.3</version>
-</dependency>
+Adding a logging framework of your choice is very straightforward. You can just add it as a dependency and it will be detected by Log4j automatically. The following example adds Log4j 2 using Gradle:
+
+```groovy
+dependencies { runtimeOnly 'org.apache.logging.log4j:log4j-core:2.11.0' }
+```
+
+You can also use and SLF4J compatible logging framework using `log4j-to-slf4j`. The following example adds Logback Classic using Gradle:
+
+```groovy
+dependencies {
+    runtimeOnly 'org.apache.logging.log4j:log4j-to-slf4j:2.11.0'
+    runtimeOnly 'ch.qos.logback:logback-classic:1.2.3'
+}
 ```
 
 ### Configure Your Logging Framework
 
-* **Logback-Classic**: [Logback configuration](https://logback.qos.ch/manual/configuration.html)
-* **Log4j 2**: [Log4j configuration](https://logging.apache.org/log4j/2.x/manual/configuration.html)
\ No newline at end of file
+* **Log4j 2**: [Log4j configuration](https://logging.apache.org/log4j/2.x/manual/configuration.html)
+* **Logback Classic**: [Logback configuration](https://logback.qos.ch/manual/configuration.html)
+
+### Logging the Relevant Shard
+
+Javacord adds the shard that is relevant for a log message. Where this is stored is called differently depending on the logging framework you are using. For Log4j 2 this is called Thread Context Map and can be added in a pattern layout with `%X{shard}`, or you can add the whole thread context map by using `%X`. For Logback Classic it is called MDC and can be added with the same pattern expressions as for Log4j.
```

Comment:
`Javacord adds the shard that is relevant for a log message` can be made more concise as `Javacord adds the relevant shard to each log message`

Reply:
`Where this is stored is called differently depending on the logging framework you are using` doesn't make that much sense, but I'm not really sure what to change it to while keeping it logically consistent. `The tag has a different name depending on which logging framework you use` or `The shard is added using a different tag depending on which logging framework you use` would be nice, but `tag` would need to be replaced with a more technically accurate name (I can't think of one off the top of my head).

Edit: Maybe just `pattern`? I'm not familiar with the terminology, so it'd be great if someone could confirm this or give a better suggestion.

Reply:
`For Log4j 2 this is called Thread Context Map` should have a comma to become `For Log4j 2, this is called Thread Context Map`. The same with the next sentence, `For Logback Classic it is called MDC` should be `For Logback Classic, it is called MDC`

Reply:
`tag` and `pattern` sound wrong technically.
How about `facility`?

Reply:
Hm, no, doesn't sound nice

Reply:
Maybe we should just say `The shard is added using a different method depending on which logging framework you use`

Reply:
Or maybe `The facility that stores this information has a different name depending on which logging framework you use.`?

Reply:
That sounds good too

Reply:
No, the method is always the same, it is just called differently by different frameworks.
It was originally called MDC, Mapped Diagnostics Context (and a similar one called NDC, Nested Diagnostics Context). Logback which is a fork of log4j1 also overtook this naming. log4j2 while supporting the old `%mdc` patterns for compatibility changed them to be ThreadContext Map and ThreadContext Stack, but they are effectively the same "thing", "facility", whatever, just called differently in different frameworks.